python/GafferSceneUI/_SceneViewInspector.py

```diff
@@ -0,0 +1,959 @@
+			if hasattr( self.__value, "value" ) :
+				self.__value = self.__value.value
+			self.__editInfo = self.__findEdit( attributeHistory, editScope )
+		else :
+			self.__editInfo = None
+
+	def value( self ) :
+
+		return self.__value
+
+	def editable( self ) :
+
+		return self.__editInfo is not None
+
+	def acquireEdit( self ) :
+
+		return self.__editInfo.acquireEdit() if self.__editInfo else None
+
+	def editType( self ) :
+
+		return self.__editInfo.editType  if self.__editInfo else None
+
+	def hasEdit( self ) :
+
+		return self.__editInfo.hasEdit if self.__editInfo else None
+
+	def inEditScope( self ) :
+
+		return self.__editInfo.inEditScope if self.__editInfo else False
+
+	def warningMessage( self ) :
+
+		return "/n".join( self.__warningMessages )
+
+	def errorMessage( self ) :
+
+		return self.__errorMessage if self.__errorMessage is not None else ""
+
+	## \todo This is a copy of TransformTool::spreadsheetAwareSource, and is
+	# eerily similar to Dispatcher::computedSource and others. This needs
+	# factoring out into some general contextAwareSource method that can be
+	# shared by all these.
+	def __spreadsheetAwareSource( self, plug ) :
+
+		plug = plug.source()
+
+		spreadsheet = plug.node()
+		if not isinstance( spreadsheet, Gaffer.Spreadsheet ) :
+			return plug
+
+		if not spreadsheet["out"].isAncestorOf( plug ) :
+			return pluge
+
+		plug = spreadsheet.activeInPlug( plug )
+		if plug.ancestor( Gaffer.Spreadsheet.RowPlug ).isSame( spreadsheet["rows"].defaultRow() ) :
+			return None
+
+		return plug.source()
+
+	def __editFromSceneNode( self, attributeHistory ) :
+
+		node = attributeHistory.scene.node()
+		if not isinstance( node, ( GafferScene.ShaderTweaks, GafferScene.Light, GafferScene.ShaderAssignment ) ) :
+			return None
+
+		if attributeHistory.scene != node["out"] :
+			return None
+
+		if isinstance( node, GafferScene.Light ) :
+
+			plug = self.__spreadsheetAwareSource( node["parameters"][ self.__parameter ] )
+			if plug is not None :
+				return _EditInfo(
+					acquireEdit = lambda : plug,
+					editType = _EditType.Source,
+					inEditScope = node.ancestor( Gaffer.EditScope ) is not None,
+					hasEdit = True
+				)
+			return None
+
+		elif isinstance( node, GafferScene.ShaderAssignment ) :
+
+			with attributeHistory.context :
+
+				if not node["filter"].getValue() & IECore.PathMatcher.Result.ExactMatch :
+					return None
+
+				shaderInput = node["shader"].getInput()
+				if shaderInput is not None :
+					plug = shaderInput.node()["parameters"].getChild( self.__parameter )
+					if plug is not None :
+						return _EditInfo(
+							acquireEdit = lambda : plug,
+							editType = _EditType.Source,
+							inEditScope = node.ancestor( Gaffer.EditScope ) is not None,
+							hasEdit = True
+						)
+				return None
+
+		elif isinstance( node, GafferScene.ShaderTweaks ) :
+
+			with attributeHistory.context :
+
+				if not node["filter"].getValue() & IECore.PathMatcher.Result.ExactMatch :
```

Comment:
Need to check `node["enabled"]` here as well.

Reply:
Checked [here](https://github.com/GafferHQ/gaffer/pull/3832/commits/c5fd1610f5e6e61763ea25870cad5942531c4d43#diff-b9d160a51cf1199c9bca6867338d8f42R450-R451).